src/backend/web/authentication.js

```diff
@@ -122,16 +122,17 @@ function protectWithRedirect(req, res, next) {
   passport.authenticate('saml')(req, res, next);
 }
 
+function throwCustomError(message, status) {
+  const err = Error(message);
+  err.status = status;
+
+  throw err;
+}
+
 // If user is not authenticated, return an appropriate 400 error type
+// eslint-disable-line no-unused-vars
 function forbidden(req, res) {
```

Comment:
I think you need to change this to:

```js
function forbidden(next) {
    const err = new Error(message);
    err.status = 403;
    next(err);
}
```

And change the callers to `forbidden()` to pass `next` vs. `req` and `res`

Reply:
We wouldn't need to `throw` in this case if using `next`?

Reply:
Like wise we'd have to update many of the function arguments which call `forbidden`, wouldn't we?

Reply:
What if we do this:

- have forbidden take all three `forbidden(req, res, next)` and the third argument.
- update all callers to pass 3 args with next as last
- rework what was here before so you do what you're doing only in the else case, and instead of sending the 403 on `res`, you called `next(err)`.  We shouldn't need to throw at all.